api/types/constants.go

```diff
@@ -952,6 +952,9 @@ const (
 
 	// ResourceSpecType refers to a resource field named "type".
 	ResourceSpecType = "type"
+
+	// ResourceSpecKind refers to a resource field named "kind".
+	ResourceSpecKind = "kind"
```

Comment:
I don't believe `Kind` is part of a resources's spec. Maybe just change to `ResourceKind`?